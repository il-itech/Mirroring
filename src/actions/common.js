import { createActions } from 'redux-actions';

export const {
  setInProgressStatus,
  setSuccessStatus,
  clearState,
  setIntervalActions,
  clearIntervalActions,
} = createActions(
  {
    SET_IN_PROGRESS_STATUS: (field, status) => ({
      field,
      status,
    }),
    SET_SUCCESS_STATUS: (field, status) => ({
      field,
      status,
    }),
    CLEAR_STATE: field => ({
      field,
    }),
  },
  'SET_INTERVAL_ACTIONS',
  'CLEAR_INTERVAL_ACTIONS',
);
