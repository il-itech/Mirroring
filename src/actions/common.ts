import { ActionCreator, createActionCreator } from 'deox';

export const setInProgressStatus = createActionCreator(
  'SET_IN_PROGRESS_STATUS',
  resolve => (field: string, status: boolean) => resolve({ field, status }),
);

export const setSuccessStatus = createActionCreator(
  'SET_SUCCESS_STATUS',
  resolve => (field: string, status: boolean) => resolve({ field, status }),
);

export const clearState = createActionCreator(
  'CLEAR_STATE',
  resolve => (field: string) => resolve({ field }),
);

export const setIntervalActions = createActionCreator(
  'SET_INTERVAL_ACTIONS',
  resolve => (actionFns: ActionCreator<string>[]) => resolve(actionFns),
);

export const clearIntervalActions = createActionCreator('CLEAR_INTERVAL_ACTIONS');
