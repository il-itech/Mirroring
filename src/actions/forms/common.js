import { createActions } from 'redux-actions';

export const {
  setFormData,
  replaceFormData,
  setFormError,
  setFormErrors,
  clearFormData,
  clearFormError,
  clearFormErrors,
  clearForm,
} = createActions(
  {
    SET_FORM_DATA: (formName, formData) => ({
      formName,
      formData,
    }),
    REPLACE_FORM_DATA: (formName, formData) => ({
      formName,
      formData,
    }),
    SET_FORM_ERROR: (formName, error) => ({
      formName,
      error,
    }),
    SET_FORM_ERRORS: (formName, errors) => ({
      formName,
      errors,
    }),
    CLEAR_FORM_DATA: formName => ({
      formName,
    }),
    CLEAR_FORM_ERROR: (formName, field) => ({
      formName,
      field,
    }),
    CLEAR_FORM_ERRORS: formName => ({
      formName,
    }),
    CLEAR_FORM: formName => ({
      formName,
    }),
  },
);
