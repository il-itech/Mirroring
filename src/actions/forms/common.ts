import { createActionCreator } from 'deox';

import { FormErrors, FormData, FieldError } from 'interfaces';

export const setFormData = createActionCreator(
  'SET_FORM_DATA',
  resolve => (formName: string, formData: FormData) => resolve({ formName, formData }),
);

export const replaceFormData = createActionCreator(
  'REPLACE_FORM_DATA',
  resolve => (formName: string, formData: FormData) => resolve({ formName, formData }),
);

export const setFormError = createActionCreator(
  'SET_FORM_ERROR',
  resolve => (formName: string, error: FieldError) => resolve({ formName, error }),
);

export const setFormErrors = createActionCreator(
  'SET_FORM_ERRORS',
  resolve => (formName: string, errors: FormErrors) => resolve({ formName, errors }),
);

export const clearFormData = createActionCreator(
  'CLEAR_FORM_DATA',
  resolve => (formName: string, field: string) => resolve({ formName, field }),
);

export const clearFormError = createActionCreator(
  'CLEAR_FORM_ERROR',
  resolve => (formName: string, field: string) => resolve({ formName, field }),
);

export const clearFormErrors = createActionCreator(
  'CLEAR_FORM_ERRORS',
  resolve => (formName: string) => resolve({ formName }),
);

export const clearForm = createActionCreator(
  'CLEAR_FORM',
  resolve => (formName: string) => resolve({ formName }),
);
