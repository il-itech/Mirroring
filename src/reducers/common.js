import * as R from 'ramda';

import {
  setInProgressStatus,
  setSuccessStatus,
  clearState,
} from 'actions/common';

export const getInitialState = (additionalState) => ({
  isInProgress: false,
  isSuccess: false,
  ...additionalState,
});

const reducerChecker = (reducerFieldName, state, fieldName, result) =>
  R.equals(reducerFieldName, fieldName)
    ? { ...state, ...result }
    : state;

export const getCommonReducers = (reducerFieldName, additionalState) => ({
  [setInProgressStatus]: (state, { payload: { field, status } }) =>
    reducerChecker(reducerFieldName, state, field, { isInProgress: status }),
  [setSuccessStatus]: (state, { payload: { field, status } }) =>
    reducerChecker(reducerFieldName, state, field, { isSuccess: status }),
  [clearState]: (state, { payload: { field } }) =>
    reducerChecker(reducerFieldName, state, field, getInitialState(additionalState)),
});
