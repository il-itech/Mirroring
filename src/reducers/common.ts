import { ReducerMap } from 'redux-actions';
import * as R from 'ramda';

import {
  setInProgressStatus,
  setSuccessStatus,
  clearState,
} from 'actions/common';
import {
  setFormData,
  replaceFormData,
  setFormError,
  setFormErrors,
  clearFormData,
  clearFormErrors,
  clearFormError,
  clearForm,
} from 'actions/forms/common';
import { ICommon, IFormCommon } from 'interfaces/state.interfaces/common-interface';

interface ICommonPayload {
  field: string;
  status?: boolean;
}
interface ICommonFormPayload {
  formName: string;
  field: string;
  formData: {};
  error?: {};
  errors?: {};
}

export const getInitialState = <T extends {}>(additionalState: T): T & ICommon => ({
  isInProgress: false,
  isSuccess: false,
  ...additionalState,
});

export const getInitialFormState = <
  T extends {}
>
  (additionalState: T): T & IFormCommon & ICommon => ({
    formData: {},
    errors: {},
    ...additionalState,
    ...getInitialState({}),
  });

const reducerChecker = <
  T extends {},
  V extends {}
>(
    reducerFieldName: string,
    state: T,
    fieldName: string,
    result: V,
  ): T | T & V =>
    R.equals(reducerFieldName, fieldName)
      ? { ...state, ...result }
      : state;

export const getCommonReducers = <
  T extends {},
  V extends {}
>(
    reducerFieldName: string,
    additionalState: V,
  ): ReducerMap<T, ICommonPayload> => ({
    [`${setInProgressStatus}`]: (state, { payload: { field, status } }): T | T & V =>
      reducerChecker(reducerFieldName, state, field, { isInProgress: status }),
    [`${setSuccessStatus}`]: (state, { payload: { field, status } }): T | T & V =>
      reducerChecker(reducerFieldName, state, field, { isSuccess: status }),
    [`${clearState}`]: (state, { payload: { field } }): T | T & V =>
      reducerChecker(reducerFieldName, state, field, getInitialState(additionalState)),
  });

export const getFormsCommonReducer = <
  T extends { formData: {}; errors: {} },
  V extends {}
>(reducerFormName: string, additionalState: V): ReducerMap<T, ICommonFormPayload> => ({
    [`${setFormData}`]: (state, { payload: { formName, formData } }): T | T & V =>
      reducerChecker(reducerFormName, state, formName, { formData: { ...state.formData, ...formData } }),

    [`${replaceFormData}`]: (state, { payload: { formName, formData } }): T | T & V =>
      reducerChecker(reducerFormName, state, formName, { formData }),

    [`${setFormError}`]: (state, { payload: { formName, error } }): T | T & V =>
      reducerChecker(reducerFormName, state, formName, { errors: { ...state.errors, ...error } }),

    [`${setFormErrors}`]: (state, { payload: { formName, errors } }): T | T & V =>
      reducerChecker(reducerFormName, state, formName, { errors }),

    [`${clearFormData}`]: (state, { payload: { formName, field } }): T | T & V =>
      reducerChecker(reducerFormName, state, formName, { formData: R.dissoc(field, state.formData) }),

    [`${clearFormError}`]: (state, { payload: { formName, field } }): T | T & V =>
      reducerChecker(reducerFormName, state, formName, { errors: R.dissoc(field, state.errors) }),

    [`${clearFormErrors}`]: (state, { payload: { formName } }): T | T & V =>
      reducerChecker(reducerFormName, state, formName, { errors: getInitialFormState(additionalState).errors }),

    [`${clearForm}`]: (state, { payload: { formName } }): T | T & V =>
      reducerChecker(reducerFormName, state, formName, getInitialFormState(additionalState)),

    ...getCommonReducers(reducerFormName, additionalState),
  });
