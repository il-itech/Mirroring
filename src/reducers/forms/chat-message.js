import { handleActions } from 'redux-actions';

import { FORM_TYPES } from 'constants';
import { getFormsCommonReducer, getInitialState } from './common';

const additionalState = {
  formData: {},
};

export const chatMessage = handleActions(
  {
    ...getFormsCommonReducer(FORM_TYPES.CHAT_MESSAGE, additionalState),
  },
  getInitialState(additionalState),
);
