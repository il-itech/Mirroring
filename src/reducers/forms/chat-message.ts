import { handleActions } from 'redux-actions';

import { IChatMessageForm } from 'interfaces/state.interfaces/forms.interfaces/form-chat-message.interface';
import { FORM_TYPES } from 'enums';
import { getFormsCommonReducer, getInitialFormState } from '../common';

const additionalState = {
  formData: {},
};

export const chatMessage = handleActions<IChatMessageForm, any>(
  {
    ...getFormsCommonReducer(FORM_TYPES.CHAT_MESSAGE, additionalState),
  },
  getInitialFormState(additionalState),
);
