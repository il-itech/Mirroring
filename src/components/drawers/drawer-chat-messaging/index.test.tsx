import { createRef } from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { chatMessages } from '__tests__/common/chat-message';
import { chatUser, chatUsers } from '__tests__/common/chat-user';
import { DrawerChatMessaging } from '.';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(fn => fn()),
  useDispatch: jest.fn(),
}));

const handlers = {
  handleSubmit: jest.fn(),
  handleKeyPress: jest.fn(),
};

const props = {
  roomId: '123qwe',
  profileId: 'profileId',
  messages: chatMessages,
  allUserList: chatUsers,
  currentChatUser: chatUser,
  chatContentRef: createRef<HTMLDivElement>(),
  ...handlers,
};

describe('<DrawerChatMessaging />', () => {
  let wrapper: ShallowWrapper;

  beforeEach(() => {
    wrapper = shallow(
      <DrawerChatMessaging {...props} />,
    ).dive();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  afterAll(() => {
    jest.restoreAllMocks();
  });

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // test('handleToggleDrawer should be called', () => {
  //   wrapper.find({ 'data-test-id': 'Button' }).simulate('click');

  //   expect(props.handleToggleDrawer).toHaveBeenCalled();
  // });
});
