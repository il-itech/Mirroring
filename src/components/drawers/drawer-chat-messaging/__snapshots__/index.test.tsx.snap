// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DrawerChatMessaging /> should match snapshot 1`] = `<Fragment />`;
